CONTRIBUTING.md

```diff
@@ -0,0 +1,93 @@
+helps prevent duplication of effort,
+and ensures that the idea fits inside the goals for the project and tools.
+It also checks that the design is sound before code is written;
+the code review tool is not the place for high-level discussions.
+
+## Testing redux
+
+Before sending code out for review, run all the tests for the whole
+tree to make sure the changes don't break other usage and keep the compatibility on upgrade:
+
+After running for a while, the command should print
+
+```
+ALL TESTS PASSED
+```
+
+## Code review
+
+Changes to Gitea must be reviewed before they are accepted,
+no matter who makes the change even if an owners or a maintainer.
+We use github's pull request workflow to do that and use [lgtm](http://lgtm.co) to ensure every PR is reviewed by at least 2 maintainers.
+
+## Contributors
+
+Everyone who sent a PR to Gitea that gets accepted will be as a contributor. Please send a PR to add your name to
+[CONTRIBUTORS](CONTRIBUTORS). For the format, see the [CONTRIBUTORS](CONTRIBUTORS).
+
+## Maintainers
+
+To make sure every PR have been checked, we make a team maintainers. Any PR MUST be reviewed and by at least two maintainers before it can get merged.
+Maintainers should be a contributor of gitea(or gogs) and contributed at least 4 accepted PRs. And a contributor should apply as a maintainer in [gitter Gitea develop](https://gitter.im/go-gitea/develop).
+And the owners or the team maintainer could invite the contributor. A maintainer should spend some time on code reviews. If some maintainer have no time
+to do that, he should apply to leave maintainers team and we will give him an honor to be as a member of advisor team. Of course, if an advisor have time to code view, welcome it back to maintainers team.
+If some one have no time to code view and forget to leave the maintainers, the owners have the power to move him from maintainers team to advisors team.
+
+## Owners
+
+Since Gitea is a pure community organization without any company support, to keep the development healthly We will elect the owners every year. Every time we will elect three owners.
+All the contributers could vote for three owners, one is the main owner, the other two are assistant owners. When the new owners have been elected, the old owners MUST move the power to the new owners. 
+If some owner don't obey these rules, the other owners are allowed to revoke his owner status.
+
+After the election, the new owners should say he agrees with these rules on the [CONTRIBUTING](CONTRIBUTING.md) on the [Gitter Gitea Channel](https://gitter.im/go-gitea/gitea). Below is the word to speak
+
+```
+I'm glad to be an owner of Gitea, I agree with [CONTRIBUTING](CONTRIBUTING.md). I will spend part of my time on gitea and lead the development of gitea.
+```
+
+For a honor to the owners, this document will add the history owners below:
+
+2016-11-04 ~ 2017-12-31 lunny<xiaolunwen@gmail.com> tboerger<thomas@webhippie.de> bkcsoft<kim.carlbacker@gmail.com>
```

Comment:
I hope one day we can get away from github, so wouldn't add more dependencies on it